desc/objectives/_generic.py

```diff
@@ -399,11 +400,16 @@ def build(self, use_jit=True, verbose=1):
             Level of output.
 
         """
-        eq = self.things[0]
+        thing = self.things[0]
         if self._grid is None:
-            grid = QuadratureGrid(eq.L_grid, eq.M_grid, eq.N_grid, eq.NFP)
-        else:
-            grid = self._grid
+            errorif(
+                self._p != "desc.equilibrium.equilibrium.Equilibrium",
+                ValueError,
+                "grid must be supplied for things besides an Equilibrium.",
+            )
+            self._grid = QuadratureGrid(
```

Comment:
does grid need to be an attribute of self?

Reply:
Sort of. The `_Objective` class does not have a grid, so we always set the user-input grid as an attribute of self to pass it from the `__init__` method to the `build` method. I will change it to `grid = self._grid` for local use in the `build` method because I think that's what we do in other objectives. But the only difference is do we want `self._grid` to return `None` or the actual default value?